Tests/AblyChatTests/DefaultRoomTests.swift

```diff
@@ -11,187 +11,155 @@ struct DefaultRoomTests {
+        let room = try await DefaultRoom(realtime: realtime, chatAPI: ChatAPI(realtime: realtime), roomID: "basketball", options: .init(), logger: TestLogger(), lifecycleManagerFactory: MockRoomLifecycleManagerFactory())
 
         // Then
         #expect(room.messages.channel.name == "basketball::$chat::$chatMessages")
     }
 
     // MARK: - Attach
 
-    @Test
-    func attach_attachesAllChannels_andSucceedsIfAllSucceed() async throws {
-        // Given: a DefaultRoom instance with ID "basketball", with a Realtime client for which `attach(_:)` completes successfully if called on the following channels:
-        //
-        //  - basketball::$chat::$chatMessages
-        //  - basketball::$chat::$typingIndicators
-        //  - basketball::$chat::$reactions
+    @Test(
+        arguments: [
+            .success(()),
+            .failure(ARTErrorInfo.createUnknownError() /* arbitrary */ ),
+        ] as[Result<Void, ARTErrorInfo>]
+    )
+    func attach(managerAttachResult: Result<Void, ARTErrorInfo>) async throws {
+        // Given: a DefaultRoom instance
         let channelsList = [
             MockRealtimeChannel(name: "basketball::$chat::$chatMessages", attachResult: .success),
-            MockRealtimeChannel(name: "basketball::$chat::$typingIndicators", attachResult: .success),
-            MockRealtimeChannel(name: "basketball::$chat::$reactions", attachResult: .success),
         ]
         let channels = MockChannels(channels: channelsList)
         let realtime = MockRealtime.create(channels: channels)
-        let room = try await DefaultRoom(realtime: realtime, chatAPI: ChatAPI(realtime: realtime), roomID: "basketball", options: .init(), logger: TestLogger())
 
-        let subscription = await room.onStatusChange(bufferingPolicy: .unbounded)
-        async let attachedStatusChange = subscription.first { $0.current == .attached }
+        let lifecycleManager = MockRoomLifecycleManager(attachResult: managerAttachResult)
+        let lifecycleManagerFactory = MockRoomLifecycleManagerFactory(manager: lifecycleManager)
 
-        // When: `attach` is called on the room
-        try await room.attach()
+        let room = try await DefaultRoom(realtime: realtime, chatAPI: ChatAPI(realtime: realtime), roomID: "basketball", options: .init(), logger: TestLogger(), lifecycleManagerFactory: lifecycleManagerFactory)
 
-        // Then: `attach(_:)` is called on each of the channels, the room `attach` call succeeds, and the room transitions to ATTACHED
-        for channel in channelsList {
-            #expect(channel.attachCallCounter.isNonZero)
+        // When: `attach()` is called on the room
+        let result = await Result { () async throws(ARTErrorInfo) in
+            do {
+                try await room.attach()
+            } catch {
+                // swiftlint:disable:next force_cast
+                throw error as! ARTErrorInfo
```

Comment:
Doesn't room already throws ARTErrorInfo? Why just re-throw it?

Reply:
Yes, it throws `ARTErrorInfo` but we don't currently communicate this in the type system (see #43), hence this cast.

Reply:
@umair-ably asked me to resolve this one so that we could merge it